.github/workflows/release.yml

```diff
@@ -3,15 +3,100 @@ name: Release new version
 on:
   workflow_dispatch:
     inputs:
-      version:
-        description: Semantic version to release
-        required: true
-        type: string
+      major-release:
+        required: false
+        description: 'Trigger a major release (optional). Leave empty for regular release.'
 
 jobs:
-  release:
-    uses: ./.github/workflows/release-workflow.yml
-    with:
-      branch: main
-      version: ${{ inputs.version }}
+  integration-tests:
+    uses: ./.github/workflows/integration-tests.yml
     secrets: inherit
+    with:
+      environment: dev
+      branch: ${{ inputs.branch }}
+
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    needs: integration-tests
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.RELEASE_PAT }}
+          ref: ${{ inputs.branch }}
+
+      - name: Setup git config
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+        shell: bash
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install argparse build semver twine
+        shell: bash
+
+      - name: Checkout release action repo
+        uses: actions/checkout@v2
+        with:
+          repository: firebolt-db/action-python-release
+          path: release_action
+
+      - name: Generate new version tag
+        id: tag_generation
+        run: |
+          OLD_TAG=$(git describe --tags --abbrev=0)
+          OLD_VERSION=$(echo $OLD_TAG | cut -c 2-)
+          echo "Old tag was ${OLD_TAG}"
+          CHANGE_LOG=$(git log $OLD_TAG..HEAD --pretty=format:%s)
+          NEW_VERSION=$(python3 release_action/scripts/generate_version_tag.py "${CHANGE_LOG}" $OLD_VERSION --major_release "${{ inputs.major-release }}")
+          NEW_TAG="v"$NEW_VERSION
+          echo "new_tag=$NEW_TAG" >>  $GITHUB_OUTPUT
+          echo "new_version=$NEW_VERSION" >>  $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Version bump
+        run: |
+          sed -i "s/^    <FileVersion>.*<\/FileVersion>/    <FileVersion>${{ steps.tag_generation.outputs.new_version }}<\/FileVersion>/" FireboltNETSDK/FireboltDotNetSdk.csproj
+          sed -i "s/^    <AssemblyVersion>.*<\/AssemblyVersion>/    <AssemblyVersion>${{ steps.tag_generation.outputs.new_version }}<\/AssemblyVersion>/" FireboltNETSDK/FireboltDotNetSdk.csproj
+          sed -i "s/^    <Version>.*<\/Version>/    <Version>${{ steps.tag_generation.outputs.new_version }}<\/Version>/" FireboltNETSDK/FireboltDotNetSdk.csproj
+          git branch
+          git diff
+          git add FireboltNETSDK/FireboltDotNetSdk.csproj
+          git commit -m "Automatic version bump to ${{ inputs.version }}"
+          git push origin ${{ inputs.branch }}
```

Comment:
Same here, `inputs.branch` is not defined. We can probably hardcode main, since we're not releasing another branch.